xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h

```diff
@@ -374,3 +375,35 @@ inline void QueueNotification(QueueMsg type, const std::string& header,
 }
 } /* namespace kodi */
 //------------------------------------------------------------------------------
+
+//============================================================================
+namespace kodi {
+///
+/// \ingroup cpp_kodi
+/// @brief Get the MD5 digest of the given text
+///
+/// @param[in]  text  text to compute the MD5 for
+/// @return           Returned MD5 digest
+///
+///
+/// ------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/General.h>
+/// ...
+/// std::string md5 = kodi::GetMD5("Make me as md5");
+/// fprintf(stderr, "My md5 digest is: '%s'\n", md5.c_str());
+/// ...
+/// ~~~~~~~~~~~~~
+///
+inline std::string GetMD5(const std::string& text)
+{
+  char* md5ret = static_cast<char*>(malloc(40*sizeof(char))); // md5 size normally 32 bytes
```

Comment:
You could add a fixed size array here to avoid `malloc` and `free`. This could be done with char `md5ret[40]`.